### PR TITLE
GEN-202: Deprecate `error_stack::Result` and `error_stack::Context`

### PR DESCRIPTION
**Pull request by TimDiekmann**
### `Context`

`core::error::Error` is stabilized and we don't need an extra trait which is doing the same thing.

### `Result`

`error_stack::Result<T, C>` is an alias to `core::result::Report<T, Report<C>>` and sometimes saves one word when typing. However, as a Rust developer one is _very_ used to `Result` and if `Result` behaves differently or has implications, this is not good. In comparison to other libraries such as `anyhow`, `error_stack` requires the context to be set as the `Err` variant, it replaces the `Result` instead of extending it. `error-stack` prefers an explicit API interface and the `Result` alias does not help.
In addition, it's almost not possible to copy code from `error-stack` as in regular cases the target still use `core::result::Result` or a mixture.

We're going to require `as _` syntax for traits in the very near future. To avoid touching the code twice and keep this PR clean, I pinned `error-stack` in the workspace to the current commit on main. 

### 🚢 Has this modified a publishable library?

This PR:

- [x] modifies a **Cargo**-publishable library, but **it is not yet ready to publish**

### 📜 Does this require a change to the docs?

The changes in this PR:

- [x] require changes to docs which **are made** as part of this PR

### 🕸️ Does this require a change to the Turbo Graph?

The changes in this PR:

- [x] do not affect the execution graph

### CODE REVIEW

File: libs/error-stack/src/context.rs [hunk 1]
```diff
@@ -56,6 +58,7 @@ use crate::Report;
 /// # assert!(err.contains::<io::Error>());
 /// # assert!(err.contains::<ConfigError>());
 /// ```
+#[deprecated(note = "Use `core::error::Error` instead", since = "0.6.0")]
```
Comment (indietyp) [inline libs/error-stack/src/context.rs:61]: should we say instead `Use core::error::Error + Send + Sync + 'static` instead? Because that's the newly required trait bound.

File: libs/error-stack/src/fmt/mod.rs [hunk 2]
```diff
@@ -754,7 +756,7 @@ fn collect<'a>(root: &'a Frame, prefix: &'a [&Frame]) -> (Vec<&'a Frame>, &'a [F
     (stack, next)
 }
 
-/// Partition the tree, this looks for the first `Context`,
+/// Partition the tree, this looks for the first `ontext`,
```
Comment (indietyp) [inline libs/error-stack/src/fmt/mod.rs:759]:
```suggestion
/// Partition the tree, this looks for the first context frame,
```

File: libs/error-stack/src/macros.rs [hunk 3]
```diff
@@ -354,14 +359,14 @@ macro_rules! bail {
 /// impl fmt::Display for PermissionDenied {
 ///     # #[allow(unused_variables)]
 ///     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
-///         const _: &str = stringify! {
+///         # const _: &str = stringify! {
```
Comment (indietyp) [inline libs/error-stack/src/macros.rs:362]: nice drive-by!

**Review by indietyp — CHANGES_REQUESTED** [review]
Thank you! Just some minor comments. 🎉

File: libs/error-stack/src/context.rs (hunk 1 above)
Comment (TimDiekmann) [reply libs/error-stack/src/context.rs:61]: In most cases the user just implements `Context`. Only if you want to be generic over the error you need those bounds. I think this is good enough.

**Review by indietyp — APPROVED** [review]
